### PR TITLE
[Sphinx] migrate introduction

### PR DESCRIPTION
**Pull request by maximedenes**
We migrate the refman introduction from LaTeX to Sphinx and integrate a few tweaks to the Sphinx infrastructure.

An independent renaming commit was added so that we can preserve history from the old manual.

### CODE REVIEW

**Comment by maximedenes** [comment]
Pipeline here: https://gitlab.com/maximedenes/coq/pipelines/18741075 (with Sphinx doc as artifact).

**Comment by maximedenes** [comment]
Rendered here: https://maximedenes.gitlab.io/-/coq/-/jobs/56878220/artifacts/_install_ci/share/doc/coq/sphinx/html/index.html

**Review by gares — APPROVED** [review]

File: doc/sphinx/introduction.rst [hunk 1]
```diff
@@ -1,143 +1,119 @@
...
-
-%The system \Coq\ is designed to develop mathematical proofs. It can be
-%used by mathematicians to develop mathematical theories and by
-%computer scientists to write formal specifications,
-The \Coq{} system is designed to develop mathematical proofs, and
+------------------------
+Introduction
+------------------------
+
+This document is the Reference Manual of version of the |Coq|  proof
```
Comment (Zimmi48) [inline doc/sphinx/introduction.rst]: Missing version. Maybe just remove "of version"?

File: doc/tools/coqrst/coqdomain.py [hunk 2]
```diff
@@ -807,6 +858,7 @@ def setup(app):
     app.connect('doctree-resolved', simplify_source_code_blocks_for_latex)
 
     # Add extra styles
+    app.add_stylesheet("fonts.css")
```
Comment (vbgl) [inline doc/tools/coqrst/coqdomain.py]: It seems that this `fonts.css` file is not in the repository. Is it an omission?

**Pull request merged**